src/ddmd/backend/elfobj.c

```diff
@@ -1749,27 +1729,26 @@ STATIC void setup_comdat(Symbol *s)
         }
         else
             *pidx = groupnamidx;
-        IDXSEC groupidx = elf_newsection2(groupnamidx, SHT_GROUP, 0, 0, 0, 0, SHN_SYMTAB, info, 4, 4);
-        int groupseg = elf_getsegment2(groupidx, info, 0);
-        seg_data *pseg = SegData[groupseg];
-        pseg->SDbuf->write32(GRP_COMDAT);
-        pseg->SDbuf->write32(comdatidx);
-
-        /* Create a section with the SHF_GROUP bit set
-         *    Section 5  .text._Z3barIiEiT_  PROGBITS,ALLOC,EXEC,GROUP,SIZE=0x000E(14),OFFSET=0x0060,ALIGN=16
-         */
+        const IDXSEC groupsecidx = elf_newsection2(groupnamidx, SHT_GROUP, 0, 0, 0, 0, SHN_SYMTAB, 0, sizeof(IDXSYM), sizeof(IDXSYM));
+        const int groupseg = elf_getsegment2(groupsecidx, 0, 0);
+        SegData[groupseg]->SDbuf->write32(GRP_COMDAT);
+
+        const char *p = cpp_mangle(s);
+        // Create a section for the comdat symbol with the SHF_GROUP bit set
         s->Sseg = ElfObj::getsegment(".text.", p, SHT_PROGBITS, SHF_ALLOC|SHF_EXECINSTR|SHF_GROUP, align);
-        IDXSEC comdatidx2 = MAP_SEG2SECIDX(s->Sseg);
-        if (comdatidx2 != comdatidx)
-        {
-            /* This can happen if there are two function definitions with the same signature.
-             * It really should be an error, but it's in the test suite as compile1.d, bug6720()
```

Comment:
Turns out the issue with [Higgs](https://ci.dlang.io/blue/organizations/jenkins/dlang-org%2Fdmd/detail/PR-6717/1/pipeline/#step-464-log-6) is caused by [Issue 17339 – ambiguous mangling with module level const values as alias template argument](https://issues.dlang.org/show_bug.cgi?id=17339). So there are 2 instantiations with identical mangling, causing `s->Sseg` to go into the section of the first instantion, and the linker complaining that the group references an earlier section.
Since they have the same mangling (and hopefully definition), and which one gets chosen depends on weak linkage anyhow, we might just reuse the first instantiations group.